sdk/servicebus/service-bus/src/util/atomXmlHelper.ts

```diff
@@ -39,51 +66,57 @@ export interface AtomXmlSerializer {
  */
 export async function executeAtomXmlOperation(
   serviceBusAtomManagementClient: ServiceClient,
-  webResource: WebResource,
+  request: PipelineRequest,
   serializer: AtomXmlSerializer,
-  operationOptions: OperationOptions
-): Promise<HttpOperationResponse> {
-  if (webResource.body) {
-    const content = serializer.serialize(webResource.body);
-    webResource.body = stringifyXML(content, { rootName: "entry" });
+  operationOptions: OperationOptions,
+  requestObject?:
+    | InternalQueueOptions
+    | InternalTopicOptions
+    | InternalSubscriptionOptions
+    | CreateRuleOptions
+): Promise<FullOperationResponse> {
+  if (requestObject) {
+    request.body = stringifyXML(serializer.serialize(requestObject), { rootName: "entry" });
   }
-
-  if (webResource.method === "PUT") {
-    webResource.headers.set("content-length", Buffer.byteLength(webResource.body));
+  if (request.method === "PUT") {
+    if (requestObject) {
```

Comment:
This check is already made once above. Can we move the PUT check and setting headers under the previous if block?